src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,115 @@
+        testUser.setAge(29);
+        REGISTRATION_SERVICE.register(testUser);
+        User testUser2 = new User();
+        testUser2.setLogin("WormJim");
+        testUser2.setPassword("654321");
+        testUser2.setAge(24);
+        assertThrows(ValidationException.class, () -> {
+            REGISTRATION_SERVICE.register(testUser2);
+        });
+    }
+
+    @Test
+    void register_invalidLoginLength_notOk() {
+        User testUser = new User();
+        testUser.setLogin("Jo");
+        testUser.setPassword("123456");
+        testUser.setAge(24);
+        assertThrows(ValidationException.class, () -> {
+            REGISTRATION_SERVICE.register(testUser);
+        });
+        testUser.setLogin("");
+        assertThrows(ValidationException.class, () -> {
+            REGISTRATION_SERVICE.register(testUser);
+        });
+        testUser.setLogin("1");
+        assertThrows(ValidationException.class, () -> {
+            REGISTRATION_SERVICE.register(testUser);
+        });
+    }
+
+    @Test
+    void register_invalidPasswordLength_notOk() {
+        User testUser = new User();
+        testUser.setLogin("Snowman");
+        testUser.setPassword("123");
+        testUser.setAge(32);
+        assertThrows(ValidationException.class, () -> {
+            REGISTRATION_SERVICE.register(testUser);
+        });
+        testUser.setPassword("");
+        assertThrows(ValidationException.class, () -> {
+            REGISTRATION_SERVICE.register(testUser);
+        });
+    }
+
+    @Test
+    void register_invalidAge_notOk() {
+        User testUser = new User();
+        testUser.setLogin("Snowman");
+        testUser.setPassword("123456");
+        testUser.setAge(17);
+        assertThrows(ValidationException.class, () -> {
+            REGISTRATION_SERVICE.register(testUser);
+        });
+        testUser.setAge(-1);
+        assertThrows(ValidationException.class, () -> {
+            REGISTRATION_SERVICE.register(testUser);
+        });
+        testUser.setAge(0);
+        assertThrows(ValidationException.class, () -> {
+            REGISTRATION_SERVICE.register(testUser);
+        });
+    }
+
+    @Test
+    void register_nullValue_notOk() {
+        User testUser = new User();
+        testUser.setLogin(null);
+        testUser.setPassword("123456");
+        testUser.setAge(18);
+        assertThrows(ValidationException.class, () -> {
+            REGISTRATION_SERVICE.register(testUser);
+        });
+        testUser.setLogin("Snowman");
+        testUser.setPassword(null);
+        assertThrows(ValidationException.class, () -> {
+            REGISTRATION_SERVICE.register(testUser);
+        });
+        testUser.setPassword("123456");
+        testUser.setAge(null);
+        assertThrows(ValidationException.class, () -> {
+            REGISTRATION_SERVICE.register(testUser);
+        });
+        assertThrows(ValidationException.class, () -> {
+            REGISTRATION_SERVICE.register(null);
+        });
```

Comment:
The test case `register_nullValue_notOk` is comprehensive and covers all the scenarios where a null value could be passed. Good job!